migrations/cockroach/V1.47__es_previous_agg_root_seq.sql

```diff
@@ -0,0 +1,377 @@
+)
+UPDATE 
+    eventstore.events 
+SET 
+    previous_aggregate_type_sequence = sp.agg_type_seq 
+FROM 
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- policy.password.complexity
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'policy.password.complexity' GROUP BY aggregate_type)
+UNION ALL
+    (SELECT 
+        e.aggregate_type, e.event_sequence, sp.agg_seq
+    FROM 
+        eventstore.events e
+    JOIN set_previous sp 
+    ON
+        e.aggregate_type = sp.agg_type
+    AND 
+        e.event_sequence > sp.agg_seq
+    ORDER BY e.event_sequence ASC
+    LIMIT 1)
+)
+UPDATE 
+    eventstore.events 
+SET 
+    previous_aggregate_type_sequence = sp.agg_type_seq 
+FROM 
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- project
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'project' GROUP BY aggregate_type)
+UNION ALL
+    (SELECT 
+        e.aggregate_type, e.event_sequence, sp.agg_seq
+    FROM 
+        eventstore.events e
+    JOIN set_previous sp 
+    ON
+        e.aggregate_type = sp.agg_type
+    AND 
+        e.event_sequence > sp.agg_seq
+    ORDER BY e.event_sequence ASC
+    LIMIT 1)
+)
+UPDATE 
+    eventstore.events 
+SET 
+    previous_aggregate_type_sequence = sp.agg_type_seq 
+FROM 
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- user
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'user' GROUP BY aggregate_type)
+UNION ALL
+    (SELECT 
+        e.aggregate_type, e.event_sequence, sp.agg_seq
+    FROM 
+        eventstore.events e
+    JOIN set_previous sp 
+    ON
+        e.aggregate_type = sp.agg_type
+    AND 
+        e.event_sequence > sp.agg_seq
+    ORDER BY e.event_sequence ASC
+    LIMIT 1)
+)
+UPDATE 
+    eventstore.events 
+SET 
+    previous_aggregate_type_sequence = sp.agg_type_seq 
+FROM 
+    set_previous sp 
+WHERE 
+    event_sequence = sp.agg_seq;
+
+
+-- --------------------------------------------------------
+-- user.email
+-- --------------------------------------------------------
+WITH RECURSIVE set_previous (agg_type, agg_seq, agg_type_seq) AS (
+    (SELECT aggregate_type, MIN(event_sequence), NULL::INT8 FROM eventstore.events WHERE aggregate_type = 'user.email' GROUP BY aggregate_type)
```

Comment:
there's no user.email aggregate type

Reply:
not anymore but there was one in the past